src/Database/Connection.php

```diff
@@ -956,11 +963,26 @@ public function getLogger()
     {
         if ($this->_logger === null) {
             $this->_logger = new QueryLogger();
+            $this->_logger->setConnectionId($this->getConnectionId());
         }
 
         return $this->_logger;
     }
 
+    /**
+     * Gets the database connection id.
+     *
+     * @return int|null
+     */
+    public function getConnectionId()
+    {
+        if ($this->_connectionId === null) {
+            $this->_connectionId = $this->_driver->getConnectionId();
```

Comment:
I don't think this should query on get.

Reply:
There is no other way to get the connection id per my knowledge.

Reply:
This should not run the sql query in a getter.  The id should be set once.

Reply:
It is not clear what you mean. `SELECT CONNECTION_ID()` query is executed only once per connection.